### ISSUE TITLE
[BUG] Flaky `test_dropout` in RandomCorrectnessTest

### ISSUE THREAD

**Issue by goyaladitya05** [keras-team/keras#22311]
The test `RandomCorrectnessTest::test_dropout` in `keras/src/random/random_test.py` is statistically flaky across all Keras backends. It fails when all elements of the input tensor are dropped, resulting in a maximum value of 0.0, which violates the assertion _ops.max(x_res) > ops.max(x)_.

## Cause: 
The test uses a dropout rate of 0.8 on a small input of shape (3, 5) (15 elements). The probability of all elements being dropped is $0.8^{15} \approx 0.035$ (3.5%). When this happens:
1. ops.max(x) is 1.0 (all inputs are ones).
2. x_res becomes all zeros.
3. ops.max(x_res) is 0.0.
4. self.assertGreater(0.0, 1.0) fails.

## Reproduction Results
Tested for 200 runs for each backend. Failures per backend:

tensorflow: 7/200
jax: 5/200
torch: 10/200
numpy: 7/200
openvino: 6/200

```
import os
import numpy as np
import keras
from keras import ops, random

def reproduce_flakiness(backend, num_runs=200):
    os.environ["KERAS_BACKEND"] = backend
    x = ops.ones((3, 5))
    rate = 0.8
    failures = 0
    for i in range(num_runs):
        x_res = random.dropout(x, rate=rate, seed=i)
        if float(ops.convert_to_numpy(ops.max(x_res))) <= 1.0:
            failures += 1
    print(f"Backend {backend}: {failures}/{num_runs} failures")
```
